shop-ui/src/main/res/layout/fragment_products_list.xml

```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools">
+
+    <data>
+
+        <variable
+            name="viewModel"
+            type="com.maxshop.viewModel.ProductsListViewModel" />
+
+        <import type="com.maxshop.adapter.Decorations" />
+    </data>
+
+    <LinearLayout
```

Comment:
There is a huge redundant nesting. Make SwipeRefreshLayout as root instead of LinearLayout